crates/node-api/src/evm.rs

```diff
@@ -0,0 +1,73 @@
+use reth_primitives::{
+    revm::env::{fill_block_env_with_coinbase, recover_header_signer},
+    Address, Chain, ChainSpec, Header, Transaction, TransactionSignedEcRecovered, U256,
+};
+use revm_primitives::{BlockEnv, CfgEnv, SpecId, TxEnv};
+
+/// This represents the set of methods used to configure the EVM before execution.
+pub trait EvmEnvConfig {
+    /// Additional metadata that can be used when filling the transaction environment.
+    type TxMeta;
+
+    // TODO: get rid of cfg,figure it out
+    /// Fill transaction environment from a [Transaction] and the given sender address.
+    fn fill_tx_env<T>(tx_env: &mut TxEnv, transaction: T, sender: Address, meta: Self::TxMeta)
+    where
+        T: AsRef<Transaction>;
+
+    /// Fill [CfgEnv] fields according to the chain spec and given header
+    fn fill_cfg_env(
+        cfg_env: &mut CfgEnv,
+        chain_spec: &ChainSpec,
+        header: &Header,
+        total_difficulty: U256,
+    );
+
+    /// Fill transaction environment from [TransactionSignedEcRecovered].
+    fn fill_tx_env_with_recovered(
+        tx_env: &mut TxEnv,
+        transaction: &TransactionSignedEcRecovered,
+        meta: Self::TxMeta,
```

Comment:
this is likely redundant:

https://github.com/paradigmxyz/reth/blob/bc229c67c9961e7d19cc8bf0b6160df9eb62aeaf/crates/rpc/rpc/src/eth/revm_utils.rs#L82-L83